src/main/components/RequestBar.jsx

```diff
@@ -119,8 +119,14 @@ const RequestBar = (props) => {
       const sendingObj = { method: selected, mode: 'cors' };
       sendingObj.headers = { 'Content-Type': contentType };
       if (headerType !== 'NONE') sendingObj.headers[headerType] = headerKey;
+
       for (let i = 0; i < testsClone.length; i += 1) {
-        sendingObj.body = JSON.stringify(testsClone[i].payload);
+        if (sendingObj.headers['Content-Type'].includes('xml')) {
+          const jsonToXml = new Builder();
+          sendingObj.body = jsonToXml.buildObject(testsClone[i].payload);
+        } else {
+          sendingObj.body = JSON.stringify(testsClone[i].payload);
+        }
         runTest(uri, sendingObj, testsClone, i);
```

Comment:
I don’t think we’ve discussed this before, but is there a particular reason we’re cloning the tests before sending them? Not sure we need to if we’re just reading them from state and not mutating them in any way.

Reply:
Yeah I think this needs another refactor for readability.  I think the reason is that we're changing state (the `status` property of each test) in the `runTest` function (AKA modifying `testsClone` and then `setTests(testsClone)` but this isn't clear after the last refactor because they happen in different functions.

Reply:
Oh yep, that‘s totally it. Makes sense but I agree, it would be good to clarify in a future refactor. Shouldn’t block this PR from merging, though